### PR TITLE
Adding method implementations to fast span/readonlyspan 

### PR DESCRIPTION
**Pull request by ahsonkhan**
Adding some missing method definitions:

==
!=
Equals
GetHashCode
CopyTo

Adding additional array constructor throws to pass CoreFX tests.
Updating Span == operator

### CODE REVIEW

File: src/mscorlib/src/System/ReadOnlySpan.cs [hunk 1]
```diff
@@ -260,6 +312,21 @@ public bool TryCopyTo(Span<T> destination)
             SpanHelper.CopyTo<T>(ref destination.DangerousGetPinnableReference(), ref DangerousGetPinnableReference(), _length);
             return true;
         }
+
+        /// <summary>
+        /// Returns true if left and right point at the same memory and have the same length.  Note that
+        /// this does *not* check to see if the *contents* are equal.
+        /// </summary>
+        public static bool operator ==(ReadOnlySpan<T> left, ReadOnlySpan<T> right)
+        {
```
Comment (jkotas) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: Could you please update the `Span<T>` implementation to match, and delete `Equals(Span<T>)` ?

File: src/mscorlib/src/System/Span.cs [hunk 2]
```diff
@@ -211,13 +211,16 @@ public bool TryCopyTo(Span<T> destination)
         /// Returns true if left and right point at the same memory and have the same length.  Note that
         /// this does *not* check to see if the *contents* are equal.
         /// </summary>
-        public static bool operator ==(Span<T> left, Span<T> right) => left.Equals(right);
+        public static bool operator ==(Span<T> left, Span<T> right)
```
Comment (jkotas) [inline src/mscorlib/src/System/Span.cs]: Could you please also delete `Equals(Span<T>)` method?

File: src/mscorlib/src/System/ReadOnlySpan.cs [hunk 3]
```diff
@@ -73,6 +79,8 @@ public ReadOnlySpan(T[] array, int start, int length)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            if (default(T) == null && array.GetType() != typeof(T[]))
```
Comment (jkotas) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: This check should not be in `ReadOnlySpan`. If it is in the slow Span version, it should be deleted there.

**Comment by jkotas** [comment]
If you would like to also add the DangerousCreate method, the implementation should simply call the internal constructor that takes `ref`:

`DangerousCreate(object obj, ref T objectData, int length) => new Span<T>(ref objectData, length);`

File: src/mscorlib/src/System/Span.cs (hunk 2 above)
Comment (ahsonkhan) [inline src/mscorlib/src/System/Span.cs]: Should I delete Equals(ReadOnlySpan<T>) as well?

File: src/mscorlib/src/System/ReadOnlySpan.cs (hunk 3 above)
Comment (ahsonkhan) [inline src/mscorlib/src/System/ReadOnlySpan.cs]: I will remove it here, disable the test, and create an issue to address this in corefx.

**Pull request merged**